pkg/riff/commands/function_create.go

```diff
@@ -172,29 +172,35 @@ func (opts *FunctionCreateOptions) Exec(ctx context.Context, c *cli.Config) erro
 	}
 	c.Successf("Created function %q\n", function.Name)
 	if opts.Tail {
-		// cancel ctx when function becomes ready
-		ctx, cancel := context.WithCancel(ctx)
-		defer cancel()
+		// err guarded by Validate()
+		timeout, _ := time.ParseDuration(opts.WaitTimeout)
+		ctx, cancel := context.WithTimeout(ctx, timeout)
+		errChan := make(chan error, 1)
+		defer close(errChan)
 
 		go func() {
 			defer cancel()
 			err := k8s.WaitUntilReady(ctx, c.Build().RESTClient(), "functions", function)
-			if err != nil {
-				c.Errorf("Error: %s\n", err)
+			if ctx.Err() == nil {
+				errChan <- err
+			}
+		}()
+		go func() {
+			defer cancel()
+			err := c.Kail.FunctionLogs(ctx, function, cli.TailSinceCreateDefault, c.Stdout)
+			if ctx.Err() == nil {
+				errChan <- err
 			}
 		}()
 
-		// err guarded by Validate()
-		timeout, _ := time.ParseDuration(opts.WaitTimeout)
-		timer := time.AfterFunc(timeout, func() {
+		<-ctx.Done()
+		if ctx.Err() == context.DeadlineExceeded {
 			c.Errorf("Timeout after %q waiting for %q to become ready\n", opts.WaitTimeout, opts.Name)
 			c.Infof("To view status run: %s function list %s %s\n", c.Name, cli.NamespaceFlagName, opts.Namespace)
 			c.Infof("To continue watching logs run: %s function tail %s %s %s\n", c.Name, opts.Name, cli.NamespaceFlagName, opts.Namespace)
-			cancel()
-		})
-		defer timer.Stop()
-
-		return c.Kail.FunctionLogs(ctx, function, cli.TailSinceCreateDefault, c.Stdout)
+			errChan <- cli.ErrSilent
```

Comment:
I think ErrSilent should wrap the underlying error otherwise silent errors will become difficult to reason about.

Reply:
done